RuntimeUnityEditor.Core/Windows/ObjectTree/RootGameObjectSearcher.cs

```diff
@@ -25,16 +25,25 @@ public class RootGameObjectSearcher
         private bool _lastSearchComponents;
         private bool _lastSearchNames;
 
-        // -1 for all scenes
-        // todo only additive scenes option?
+        public enum SearchType
+        {
+            AllScenes,
+            DontDestroyOnLoad,
+            Inactive,
+            SceneIndex,
+        }
+
+        public SearchType CurrentSearchType { get; set; } = SearchType.AllScenes;
+
         private int _sceneIndexFilter = -1;
         internal int SceneIndexFilter
```

Comment:
SearchType and SceneIndexFilter can be combined into a single property. Set enum member `SearchType.SceneIndex = 10` (or some higher number) and simply add the scene index to SearchType.SceneIndex, so scene index `SearchType.SceneIndex + 0 == 10`, `SearchType.SceneIndex + 1 == 11` etc.